packages/dd-trace/src/config.js

```diff
@@ -1019,6 +1013,13 @@ class Config {
       calc['tracePropagationStyle.inject'] = calc['tracePropagationStyle.inject'] || defaultPropagationStyle
       calc['tracePropagationStyle.extract'] = calc['tracePropagationStyle.extract'] || defaultPropagationStyle
     }
+
+    const iastEnabled = coalesce(this._options['iast.enabled'], this._env['iast.enabled'])
+    const profilingEnabled = coalesce(this._options['profiling.enabled'], this._env['profiling.enabled'])
+    const injectionIncludesProfiler = (this._env.injectionEnabled || []).includes('profiler')
+    if (iastEnabled || ['auto', 'true'].includes(profilingEnabled) || injectionIncludesProfiler) {
```

Comment:
If I understanding this correctly... previous approach: `DD_TELEMETRY_LOG_COLLECTION_ENABLED=false` (manually set to false) with iast enabled => telemetry.logCollection == disabled

with current approach
`DD_TELEMETRY_LOG_COLLECTION_ENABLED=false` (manually set to false) with iast enabled => telemetry.logCollection == enabled

Am I right? Or am I missunderstanding something related with `_setBoolean(calc...`

Reply:
Calculated values have lower priority than environment variables  (see the order of elements in the `containers` array in `_merge()` function), so an explicit `false` in env vars should still take precedence.